### ISSUE TITLE
x/pkgsite: source information is not automatically updated if changes for an existing module version at a vanity URL

### ISSUE THREAD

**Issue by changkun** [golang/go#46203]
### What is the URL of the page with the issue?
 
 https://pkg.go.dev/golang.design/x/clipboard
 
### What is your user agent?
 

 
 Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36
 
### Screenshot
 

 
![image](https://user-images.githubusercontent.com/5498964/118489712-72081900-b71d-11eb-88d0-a8dc146509ba.png) 
 
### What did you do?
 

 
The vanity URL returned an incorrect address previously, the newly released version cannot be updated to the sum db and the document page failed to synchronize with a newly released version (0.4.6).
 
### What did you expect to see?
 
All released versions are shown and can be fetched via go get
 
### What did you see instead?

v0.4.5 and v0.4.6 are not showing on the pkgsite, and go get failed on 0.4.5

**Comment by julieqiu** [comment]
https://pkg.go.dev/golang.design/x/clipboard currently shows v0.4.6 for me. Is that what you were looking for?
 

![image](https://user-images.githubusercontent.com/5943521/118525452-31a09f00-b70d-11eb-8f33-310bc4e876c1.png)

**Comment by changkun** [comment]
Interesting, but the 0.4.5 is not included (still). Here is another which does not update the repository link on the right side of the page:

https://pkg.go.dev/golang.design/x/hotkey

**Comment by julieqiu** [comment]
It looks like https://pkg.go.dev/golang.design/x/clipboard@v0.4.5 and https://pkg.go.dev/golang.design/x/hotkey@v0.1.1 have been updated.

**Comment by changkun** [comment]
@julieqiu No. The problem remains exist. We have many other projects and all suffering the problem (still):

- https://pkg.go.dev/golang.design/x/code2img
- https://pkg.go.dev/golang.design/x/redir
- https://pkg.go.dev/golang.design/x/tgstore
- https://pkg.go.dev/golang.design/x/mainthread
- ...

The issue indicates that the database is neither fetching the real-time vanity URL information nor updates the information regularly.

**Comment by julieqiu** [comment]
@changkun thanks for following up. You are correct that once we fetch a module version, we do not refetch that information again. Since all of the modules currently exist in our database, the source information will not be re-updated. I'll reopen this issue.

**Comment by changkun** [comment]
@julieqiu Thanks for reopening. It is still somehow conflicts with your interpretation. For instance, this package:

https://pkg.go.dev/golang.design/x/verbose

The last update was on Sep 16, 2020.

![image](https://user-images.githubusercontent.com/5498964/118933367-639e4500-b949-11eb-85b2-b7422b69ccf6.png)

The page remains show `github.com/golang-design/*/verbose` on the right side.

There are many different versions in the previously mentioned packages released afterward, which means the database seems good at those moments, and the page records the correct record. However, somehow the repository information was refreshed globally.

Similar repositories remain to suffer from the issue:

- https://pkg.go.dev/golang.design/x/bench
- https://pkg.go.dev/golang.design/x/thread
- https://pkg.go.dev/golang.design/x/mkill
- https://pkg.go.dev/golang.design/x/lockfree

**Comment by julieqiu** [comment]
> There are many different versions in the previously mentioned packages released afterward, which means the database seems good at those moments, and the page records the correct record. However, somehow the repository information was refreshed globally.

We monitor the Go Module Index (https://index.golang.org/index) for new modules, and when we see a new module, we download it from https://proxy.golang.org. During our download process, we also check what the repository information is, and we store that information in our database. We do not automatically refresh this information later, which is why it will be stale if you change the repository URL that your vanity URL points to afterwards.

I manually updated the latest version for each of the links you sent me, which is why those pages are now up to date. For example, https://pkg.go.dev/golang.design/x/tgstore@v0.2.1 shows `github.com/golang-design/tgstore` as the repository URL. I didn't update any of the earlier versions, so https://pkg.go.dev/golang.design/x/tgstore@v0.1.0, for example, still shows `github.com/golang-design/*/tgstore` as the repository URL.

Hope that helps clarify things! For now, we recommend creating a new version for the remaining modules that are affected, so that the latest version on pkg.go.dev is showing the correct repository URL.
